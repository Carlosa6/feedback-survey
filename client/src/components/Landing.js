import React from 'react';
import { connect } from 'react-redux';

export const Landing = ({ loggedInUser }) => {

    return (
        <div style={{ textAlign: 'center' }}>
            <h1>Emaily</h1>
            <h5>{loggedInUser ? 'User is logged in' : 'Please login to continue'}</h5>
            <p>Come here to collect Feedback!</p>
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        loggedInUser: state.auth
    }
}

export default connect(mapStateToProps, null)(Landing);