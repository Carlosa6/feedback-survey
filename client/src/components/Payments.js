import React, { Component } from 'react';
import StripeCheckout from 'react-stripe-checkout';
import { connect } from 'react-redux';
import * as actions from '../actions/index';

export class Payments extends Component {
    render() {
        return (
            <StripeCheckout
                name="Emaily"
                description="$50 for 50 email credits"
                amount={5000}
                token={token => this.props.handleToken(token)}
                stripeKey={process.env.REACT_APP_STRIPE_KEY ? process.env.REACT_APP_STRIPE_KEY : 'test_key'}>
                <button className="btn-flat">
                    Add Credits
                </button>
            </StripeCheckout>
        )
    }
}


export default connect(null, actions)(Payments);