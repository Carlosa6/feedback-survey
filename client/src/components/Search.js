import React, { Component } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import SurveyField from './Surveys/SurveyField';
import * as actions from '../actions/index';


class Search extends Component {
    state = { searchValue: "" };
    constructor(props) {
        super(props);
        this.state = { searchValue: "" };
    }

    renderContent(surveys) {
        return surveys.map((survey, index) => {
            return (
                <div key={index}>
                    <div key={survey['title']}>
                        <label>{survey['title']}</label>
                    </div>
                    <div key={index + `body`}>
                        {survey['body']}
                    </div>
                </div>
            );
        });
    }
    render() {
        return (
            <div className="container ">
                <div className="space-space-top1 parallax">
                    <form >
                        <Field component={SurveyField} type="text" label="SEARCH" name="search" key="search" />
                        <button className="teal btn-flat center white-text" onClick={(evt) => { this.props.searchSurvey(this.props.searchValue.values['search']); evt.preventDefault(); }}>Click me!</button>
                    </form>
                </div>
                {this.props.searchData ? this.renderContent(this.props.searchData) : null}
            </div >
        );
    }
};

function mapStateToProps({ form, surveys }) {
    debugger;
    return {
        searchValue: form.searchForm,
        searchData: surveys
    }
}


export default reduxForm({
    form: 'searchForm'
})(connect(mapStateToProps, actions)(Search));