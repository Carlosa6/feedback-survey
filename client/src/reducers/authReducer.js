import { FETCH_USER, FETCH_USER_SUCCEESS } from '../actions/types';

export const initialState = {
    auth: {},
    loaded: false
};

export const authReducer = function (state = null, action) {
    switch (action.type) {
        case FETCH_USER:
            return action.payload || false;
        case FETCH_USER_SUCCEESS:
            return { payload: action.payload, loaded: true }
        default:
            return state;
    }
}

export default authReducer;