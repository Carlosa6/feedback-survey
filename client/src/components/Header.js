import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Payments from './Payments';

export class Header extends Component {


    renderContent() {
        switch (this.props.auth) {
            case null: return;
            case false: return (
                <li><a href="/auth/google">Login with google</a></li>
            );
            default: return [
                <li key="surveys"><Link to={this.props.auth ? '/surveys/new' : '/'}><button className="btn-flat" >
                    Create Surveys </button></Link></li>,
                <li key="payment"><Payments /></li>,
                <li key="credits" style={{ margin: '0 10px' }}>Credits:{this.props.auth.credits}</li>,
                <li key="logout"> <a href="/api/logout">Logout</a> </li>];
        }
    }

    routehandler() {
        this.props.history.push('/surveys/new');
    }

    render() {
        return (
            <nav>
                <div className="nav-wrapper">
                    <Link
                        to={this.props.auth ? '/surveys' : '/'}
                        className="brand-logo">Emaily</Link>
                    <ul id="nav-mobile" className="right hide-on-med-and-down">
                        {this.renderContent()}
                    </ul>
                </div>
            </nav >
        );
    }
}

function mapStatesToProp({ auth }) {
    return { auth }
}

export default connect(mapStatesToProp, null)(Header);