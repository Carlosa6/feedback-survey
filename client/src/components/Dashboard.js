import React from 'react';
import { Link } from 'react-router-dom';
import Search from './Search';

const Dashboard = () => {
    return (
        <div style={{ textAlign: 'center' }}>
            <h1>Dashboard</h1>
            <div><Search /></div>
            <div className="fixed-action-btn space-top1">
                <Link
                    style={{ position: 'absolute', bottom: '50px', right: '50px' }} to="/surveys/new" href="#" className="btn-floating btn-large red">
                    <i className="large material-icons">add</i>
                </Link>
            </div>
        </div >
    );
}

export default Dashboard;