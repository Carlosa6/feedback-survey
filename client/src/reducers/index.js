import { combineReducers } from 'redux';
import { reducer as reduxForm } from 'redux-form';
import authReducer from "./authReducer";
import dataReducer from './dataReducer';
// import dataReducer from "./dataReducer";

export default combineReducers({
    auth: authReducer,
    form: reduxForm,
    surveys: dataReducer
});