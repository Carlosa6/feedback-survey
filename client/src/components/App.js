import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import Header from './Header';
import Landing from './Landing';
import Dashboard from './Dashboard';
import SurveyNew from './Surveys/SurveyNew';

class App extends Component {

    componentDidMount() {
        this.props.fetchUser();
    }

    render() {
        return (
            <div className="container" >
                <div className="container">
                    <Header />
                    <Route exact path="/" component={Landing} />
                    <Route exact path="/surveys" component={Dashboard} />
                    <Route path="/surveys/new" component={SurveyNew} />
                </div>
            </div >
        );
    }
};

export default withRouter(connect(null, actions)(App));