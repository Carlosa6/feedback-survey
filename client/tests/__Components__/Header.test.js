import React from 'react';
import Enzyme, { shallow, mount, render } from "enzyme";
import { shallowToJson } from 'enzyme-to-json';
import { Header } from '../../src/components/Header';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from 'redux';
import reduxThunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import reducers from '../../src/reducers';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
const store = createStore(reducers, {}, applyMiddleware(reduxThunk));

function shallowSetup() {
    // Sample props to pass to our shallow render
    const props = {
        auth: {
            googleId: '1021201020102',
            credits: 500
        }
    }
    const enzymeWrapper = shallow(<Header {...props} />);
    return {
        props,
        enzymeWrapper
    };
}

describe("Header Component", () => {

    const { enzymeWrapper, props } = shallowSetup()
    beforeAll(() => {
        wrapper = mount(<MemoryRouter><Provider store={store} ><Header {...props} /></Provider></MemoryRouter>);
    });
    test('renders Header component correctly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
