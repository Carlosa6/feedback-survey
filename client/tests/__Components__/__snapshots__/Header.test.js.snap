// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders Header component correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "hg3az8",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "hg3az8",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Header
        auth={
          Object {
            "credits": 500,
            "googleId": "1021201020102",
          }
        }
      >
        <nav>
          <div
            className="nav-wrapper"
          >
            <Link
              className="brand-logo"
              to="/surveys"
            >
              <LinkAnchor
                className="brand-logo"
                href="/surveys"
                navigate={[Function]}
              >
                <a
                  className="brand-logo"
                  href="/surveys"
                  onClick={[Function]}
                >
                  Emaily
                </a>
              </LinkAnchor>
            </Link>
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
            >
              <li
                key="surveys"
              >
                <Link
                  to="/surveys/new"
                >
                  <LinkAnchor
                    href="/surveys/new"
                    navigate={[Function]}
                  >
                    <a
                      href="/surveys/new"
                      onClick={[Function]}
                    >
                      <button
                        className="btn-flat"
                      >
                        Create Surveys 
                      </button>
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
              <li
                key="payment"
              >
                <Connect(Payments)>
                  <Payments
                    fetchUser={[Function]}
                    handleToken={[Function]}
                    searchSurvey={[Function]}
                    submitSurvey={[Function]}
                  >
                    <ReactStripeCheckout
                      ComponentClass="span"
                      amount={5000}
                      className="StripeCheckout"
                      description="$50 for 50 email credits"
                      label="Pay With Card"
                      locale="auto"
                      name="Emaily"
                      reconfigureOnUpdate={false}
                      stripeKey="test_key"
                      token={[Function]}
                      triggerEvent="onClick"
                    >
                      <span
                        onClick={[Function]}
                      >
                        <button
                          className="btn-flat"
                        >
                          Add Credits
                        </button>
                      </span>
                    </ReactStripeCheckout>
                  </Payments>
                </Connect(Payments)>
              </li>
              <li
                key="credits"
                style={
                  Object {
                    "margin": "0 10px",
                  }
                }
              >
                Credits:
                500
              </li>
              <li
                key="logout"
              >
                 
                <a
                  href="/api/logout"
                >
                  Logout
                </a>
                 
              </li>
            </ul>
          </div>
        </nav>
      </Header>
    </Provider>
  </Router>
</MemoryRouter>
`;
